app/helpers/orders_helper.rb

```diff
@@ -0,0 +1,27 @@
+module OrdersHelper
+  def generate_invoice_number
+    # Logic to generate a unique invoice number, e.g., using timestamp
+    Time.now.strftime('%Y%m%d%H%M%S')
+  end
+
+  def move_cart_items_to_order(order)
+    @cart.cart_items.each do |cart_item|
+      # OrderItem.create(
+      #   order: @order,
+      #   product: cart_item.product,
+      #   product_name: cart_item.product.name,
+      #   product_description: cart_item.product.description,
+      #   price_per_unit: cart_item.product.price,
+      #   quantity: cart_item.quantity
+      # )
+      order.order_items.build(product: cart_item.product, quantity: cart_item.quantity)
```

Comment:
This is making N query to database. Try to solve this issue. (there is a concept of accept_neasted_attributes)

Reply:
Let me explore that, thanks.